common/src/main/java/com/wynntils/features/user/overlays/InfoBoxesFeature.java

```diff
@@ -0,0 +1,218 @@
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox4Overlay = new InfoBoxOverlay(4);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox5Overlay = new InfoBoxOverlay(5);
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay infoBox6Overlay = new InfoBoxOverlay(6);
+
+    private static final Pattern INFO_VARIABLE_PATTERN =
+            Pattern.compile("%([a-zA-Z_]+|%)%|\\\\([\\\\n%§EBLMH]|x[\\dA-Fa-f]{2}|u[\\dA-Fa-f]{4}|U[\\dA-Fa-f]{8})");
+
+    public static class InfoBoxOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public String content = "";
+
+        private final int id;
+
+        private TextRenderTask toRender = recalculateFunctions(content);
+        private List<Function<?>> functionDependencies = new ArrayList<>();
+
+        public InfoBoxOverlay(int id) {
+            super(
+                    new OverlayPosition(
+                            -60 + (15 * id),
+                            5,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.MiddleLeft),
+                    new GuiScaledOverlaySize(120, 10),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle);
+            this.id = id;
+        }
+
+        private TextRenderTask recalculateFunctions(String renderableText) {
+            if (functionDependencies == null) {
+                functionDependencies = new ArrayList<>();
+            }
+
+            for (Function<?> oldDependency : functionDependencies) {
+                FunctionManager.disableFunction(oldDependency);
+            }
+
+            functionDependencies.clear();
+
+            StringBuilder builder = new StringBuilder(renderableText.length() + 10);
+            Matcher m = INFO_VARIABLE_PATTERN.matcher(renderableText);
+            while (m.find()) {
+                String replacement = null;
+                if (m.group(1) != null && FunctionManager.forName(m.group(1)).isPresent()) {
+                    // %variable%
+                    Function<?> function = FunctionManager.forName(m.group(1)).get();
+
+                    FunctionManager.enableFunction(function);
+                    functionDependencies.add(function);
+
+                    replacement = FunctionManager.getRawValueString(function, "");
+                } else if (m.group(2) != null) {
+                    // \escape
+                    replacement = doEscapeFormat(m.group(2));
+                }
+                if (replacement == null) {
+                    replacement = m.group(0);
+                }
+
+                m.appendReplacement(builder, replacement);
+            }
+            m.appendTail(builder);
```

Comment:
In any case, if you do keep this functionality, it should move to FunctionManager, to something like `getStringFromLegacyTemplate()`...